src/section/member_info.rs

```diff
@@ -36,12 +36,15 @@ impl MemberInfo {
     }
 
     // Converts this info into one with the state changed to `Left`.
+    // TODO: review if we still need this function
```

Comment:
We do. The reason why it seems like we don't is because you commented out all the processing of nodes going offline.

Reply:
yeah, i just put a generic comment in these places to remove clippy issues as suggested by Qi, so whenever we re-enable the logic for nodes going offline this can be re-enabled.